gridfinity-rebuilt-utility.scad

```diff
@@ -291,15 +294,29 @@ module block_base(hole_options, off=0, size=[BASE_SIZE, BASE_SIZE]) {
             translate([translation_x, 0, 0])
             polygon(BASE_PROFILE);
 
-            rounded_square(
-                [
-                    base_bottom_size.x + TOLLERANCE,
-                    base_bottom_size.y + TOLLERANCE,
-                    BASE_PROFILE_MAX.y
-                ],
-                translation_x,
-                center=true
-            );
+            if (thumbscrew) {
+                ScrewHole(15, 5, position=[0, 0, 0], pitch=1.5) {
```

Comment:
If you'll take a look at what the `ScrewHole` module does, you'll see it's a wrapper around `diference(){children() ScrewThread(...)}` with some translations that aren't used here.

Since this is already inside a `difference()` statement, there's no need for a wrapper, and it both duplicates code and doesn't fit with how holes are handled in this function.

I'd suggest putting the if statement and a raw `ScrewThread(...)` call immediately before or after the section with the comment "4 holes".

Reply:
Thanks, I've refactored this to just do a call to ScrewHole() as suggested. Works well.